mwapi/session.py

```diff
@@ -84,66 +92,42 @@ def logout(self):
         Logs out of the session with MediaWiki
         """
         self.post(action='logout')
-        self.is_authenticated = False
```

Comment:
password auth is still here.  I think that is important.  The boolean local variable is problematic for a lot of reasons.  Essentially you don't know when it is out of date.   But yeah, I figured that a a persistent auth-cookie should be dropped and OAuth support used instead.